### PR TITLE
fix: Show only hacktoberfest topic related issues

### LINKED ISSUE #203: Take into account updated Hacktoberfest Rules
It would be nice if we could only display issues on repositories which meet the updated rules and requirements of Hacktoberfest.

[See rule updates here](https://hacktoberfest.digitalocean.com/hacktoberfest-update)

### PR DESCRIPTION
**Pull request by Kalovelo**
This PR filters the fetched issues by topic (should be hacktoberfest).

Resolves #203 

The most **efficient** solution would be to use GraphQL queries to filter those issues. However, the graphql API needs authentication. 
The `search/issues` route does not provide a way to filters issues by topic. Therefore, the only way without passing the fetching rate limit was with 2 fetching queries each time.
1. The first one fetches all the hacktoberfest topic related repos
2. The second one (the original fetch) fetches all issues based on the user's preference (language/etc).

Both fetching queries fetch at their maximum data (`per_page=100`). 

Finally, those 2 results are compared, and only the issues (2nd fetch) with an existing `repository_url` in the repos data (1st fetch)  are shown.

Sadly, due to api unauthorized limits, I didn't find a more efficient way to implement, and I believe that the (authorized) Graphql option would work nicer.

### CODE REVIEW

**Comment by damcclean** [comment]
Thanks for the pull request! I'll review it this evening after work. 

If we were to go down the route with authorisation, what would that need from me, just creating a Github key?

**Comment by Kalovelo** [comment]
> If we were to go down the route with authorisation, what would that need from me, just creating a Github key?

I am currently working on changing the fetching from REST to GraphQL :tada:

I will set `.env` variable for the github token

**Comment by damcclean** [comment]
> > If we were to go down the route with authorisation, what would that need from me, just creating a Github key?
> 
> I am currently working on changing the fetching from REST to GraphQL 🎉
> 
> I will set `.env` variable for the github token

Awesome, we may also need to add a step to the `README.md` file so that new developers coming to the project know that they'll need to generate a key as part of the setup.

**Comment by Kalovelo** [comment]
PR updated, the fetching is done with Graphql :tada:

Issues are now topic related, and a github bearer token is required to make it work.

*The rate limit of GraphQL can be found [here](https://docs.github.com/en/free-pro-team@latest/graphql/overview/resource-limitations).

File: secrets.key.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export default {
+  // https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token
```
Comment (damcclean) [inline secrets.key.js:2]: Could you remove this comment here, where to enter the key is already explained in the readme?

File: src/js/app.js [hunk 2]
```diff
@@ -1,6 +1,9 @@
+import secrets from '../../secrets.key';
```
Comment (damcclean) [inline src/js/app.js:1]: A small nitpick of mine, could you remove the `;` at the ends of the lines?

File: src/js/app.js [hunk 3]
```diff
@@ -16,55 +19,125 @@ new Vue({
...
+                    body:JSON.stringify({query:query})
+                }
+            );
+            const data = await result.json();
+            const searchedRepos = data.data.search.repos; 
+            this.cursor = searchedRepos[searchedRepos.length-1].cursor;
+            const repos = data.data.search.repos.filter(({repo})=>repo.issues.issuesAll.length>0);
+            return repos;
+        },
+        async getIssues(){
```
Comment (damcclean) [inline src/js/app.js:94]: Another nitpick (sorry): could you leave a one line gap in between each of these methods please?

File: src/js/app.js [hunk 4]
```diff
@@ -16,55 +19,125 @@ new Vue({
...
+            const result = await fetch('https://api.github.com/graphql',
+                {   method:'POST',
+                    headers:headers,
+                    body:JSON.stringify({query:query})
+                }
+            );
+            const data = await result.json();
+            const searchedRepos = data.data.search.repos; 
+            this.cursor = searchedRepos[searchedRepos.length-1].cursor;
+            const repos = data.data.search.repos.filter(({repo})=>repo.issues.issuesAll.length>0);
```
Comment (damcclean) [inline src/js/app.js:91]: Could you mind making this line a little more readable, like this:

```js
const repos = data.data.search.repos.filter((repo) => {
    return repo.issues.issuesAll.length > 0
})
```

**Review by damcclean — CHANGES_REQUESTED** [review]
I've left some comments below. I'll have to check this out again tomorrow or during the weekend to ensure everything works with a valid Github key, as I can't confirm it with Netlify.

**Comment by damcclean** [comment]
Looks good, thanks for the follow-up commit. I'll do some final testing tomorrow or during the weekend and get it merged. Thanks for your work on this one!

**Comment by damcclean** [comment]
I've merged this PR in, made a little change to store the key in an `.env` file instead. Have a nice day :)

**Comment by Kalovelo** [comment]
Sweet. you too @damcclean !

**Pull request merged**